photon_installer/installer.py

```diff
@@ -926,10 +926,28 @@ def _setup_install_repo(self):
     def _install_additional_rpms(self):
         rpms_path = self.install_config.get('additional_rpms_path', None)
 
-        if not rpms_path or not os.path.exists(rpms_path):
+        rpm_dir = []
+        if not rpms_path or len(rpm_dir) == 0:
             return
 
-        if self.cmd.run(['rpm', '--root', self.photon_root, '-U', rpms_path + '/*.rpm']) != 0:
+        if isinstance(rpms_path, list):
+            rpm_dir = [d for d in rpm_dir if len(d) > 0
+                       and os.path.exists(d)]
```

Comment:
We should throw an error if the path does not exist.

Better yet, check if the directory contains any rpm files.

Reply:
So in this patch, I indicate what the issue is closely related.
https://github.com/spyroot/photon-os-installer/tree/patch-2

The problem is that sub-process might not report the exit code. 
So if you have an exception raised. We don't know about that;  it not reported from cmd. run, 
the same issue if you have a run time exception or if the child PID interleaved. 

Since we don't read from stderr and parse.  Another issue sub-process might 
actually, execute before CDROM is even mounted.

If you check the exit code from the rpm tool, you will see it returns different exit codes
that is another issue.

Suppose RPM in additional rpm has conflict, i.e., ver. > Then, in the  RPM in actual ISO,  
if RPM is already installed, or exception raised,  or a version of rpm i
n additional < than in ISO rpms.  

For example, if generated in the script additional rpms, you always 
download python3_something. ( latest from our repo).  

Since we invoke rpm with a list if one rpm in that list is wrong based 
on condition I indicated.

And because it is a sub-process, debugging is tough for someone who wrote Kickstart. 

Essentially it boils down. Do you want a crash installer or not? If we want to crash, 
we need to dump from the sub-process stderr, read it, and serialize it to a log. 

That way, using the best-effort strategy here, IMHO better. so a person will be able to install OS
, figure out that rpms additional did not succeed,  resolve conflict ( ver/path, etc.) 
go back and fix the kickstart file.